src/test/java/seedu/address/model/person/PersonTagsContainsTagsPredicateTest.java

```diff
@@ -48,54 +48,124 @@ public void test_tagsContainTags_returnsTrue() {
-        assertFalse(fifthPredicate.test(new PersonBuilder().withName("Alice Bob").withTags("Bobs", "Carol").build()));
+        personToTest = createsPersonWithTags("Alice Bob", "Carol", "Bobs");
+        assertFalse(fifthPredicate.test(personToTest));
+    }
+    @Test
+    public void test_tagsContainOneTag_returnsTrue() {
+        // One tag
+        List<Tag> firstTags = new ArrayList<>();
+        firstTags.add(new Tag("Alice"));
+        PersonTagsContainsTagsPredicate firstPredicate = new PersonTagsContainsTagsPredicate(firstTags);
+        Person personToTest = createsPersonWithTags("Alice Bob", "Alice");
+        assertTrue(firstPredicate.test(personToTest));
+    }
+
+    @Test
+    public void test_tagsContainMultipleTags_returnsTrue() {
+
+        // Multiple tag
+        List<Tag> secondTags = new ArrayList<>();
+        secondTags.add(new Tag("Alice"));
+        secondTags.add(new Tag("Bob"));
+        PersonTagsContainsTagsPredicate secondPredicate = new PersonTagsContainsTagsPredicate(secondTags);
+        Person personToTest = createsPersonWithTags("Alice Bob", "Alice", "Bob");
+        assertTrue(secondPredicate.test(personToTest));
+    }
+    @Test
+    public void test_tagsContainOnlyOneTag_returnsFalse() {
+        // Only one matching tag
+        List<Tag> thirdTags = new ArrayList<>();
+        thirdTags.add(new Tag("Bob"));
+        thirdTags.add(new Tag("Carol"));
+        PersonTagsContainsTagsPredicate thirdPredicate = new PersonTagsContainsTagsPredicate(thirdTags);
+        Person personToTest = createsPersonWithTags("Alice Carol", "Friend", "Bob");
+        assertFalse(thirdPredicate.test(personToTest));
+    }
+
+    @Test
+    public void test_tagsContainCaseInsensitiveTags_returnsTrue() {
+        // Mixed-case tag
+        List<Tag> fourthTags = new ArrayList<>();
+        fourthTags.add(new Tag("BOb"));
+        fourthTags.add(new Tag("cAroL"));
+        PersonTagsContainsTagsPredicate fourthPredicate = new PersonTagsContainsTagsPredicate(fourthTags);
+        Person personToTest = createsPersonWithTags("Alice Bob", "Carol", "Bob");
+        assertTrue(fourthPredicate.test(personToTest));
+    }
+
+    @Test
+    public void test_tagsContainSubsetTags_returnsFalse() {
+        // Subset tag
+        List<Tag> fifthTags = new ArrayList<>();
+        fifthTags.add(new Tag("BOb"));
+        fifthTags.add(new Tag("cAroL"));
+        PersonTagsContainsTagsPredicate fifthPredicate = new PersonTagsContainsTagsPredicate(fifthTags);
+        Person personToTest = createsPersonWithTags("Alice Bob", "Carol", "Bobs");
+        assertFalse(fifthPredicate.test(personToTest));
+    }
+
+    @Test
+    public void test_personOtherAttributeHasTagString_returnsFalse() {
+        String[] tagString = new String[]{"12345", "Main", "Street"};
+        List<Tag> secondTagList = Arrays.stream(tagString).map(Tag::new).collect(Collectors.toList());
```

Comment:
Can consider standardizing how you generate the List<Tag> among different test cases
Perhaps have a helper method to create List<Tag> from strings, single level of abstraction stuff